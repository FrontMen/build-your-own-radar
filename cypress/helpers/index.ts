export const dataCy = (value: string) => cy.get(`[data-cy=${value}]`);
export const dataTestId = (value: string) => cy.get(`[data-testid=${value}]`);

export const hexToRgb = (hex: string) => {
  const bigint = parseInt(hex.replace('#', ''), 16);
  const r = (bigint >> 16) & 255;
  const g = (bigint >> 8) & 255;
  const b = bigint & 255;

  return `rgb(${r}, ${g}, ${b})`;
}
