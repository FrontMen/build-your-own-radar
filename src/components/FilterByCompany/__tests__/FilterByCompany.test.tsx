import React from 'react';
import { FilterByCompany } from '..';
import { withAllProviders } from 'test/helpers';

describe('FilterByCompany', () => {
  it('should matches the snapshot', () => {
    const { container } = withAllProviders(<FilterByCompany />);
    expect(container).toMatchSnapshot();
  });
});
