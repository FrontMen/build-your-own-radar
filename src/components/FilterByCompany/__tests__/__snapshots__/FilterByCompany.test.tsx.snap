// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FilterByCompany should matches the snapshot 1`] = `
.c3 {
  border: 2px solid #526de4;
  background-color: #526de4;
  padding: 2px;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  cursor: pointer;
  position: relative;
  width: 18px;
  height: 18px;
  font-family: Roboto,san-serif;
  font-size: 1em;
  font-weight: 400;
  line-height: 1.65;
}

.c2 {
  font-family: Roboto,san-serif;
  font-size: 1em;
  font-weight: 400;
  line-height: 1.65;
  margin-right: 16px;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  margin: 8px;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  margin: 8px 0;
}

<div>
  <div
    class="c0"
  >
    <div
      class="c1"
      data-testid="ITR_BE"
    >
      <label
        class="c2"
      >
        Intracto BE
      </label>
      <span
        class="c3"
        data-testid="checkbox"
      >
        <svg
          color="#fff"
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          stroke-width="0"
          style="color: rgb(255, 255, 255);"
          viewBox="0 0 512 512"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M173.898 439.404l-166.4-166.4c-9.997-9.997-9.997-26.206 0-36.204l36.203-36.204c9.997-9.998 26.207-9.998 36.204 0L192 312.69 432.095 72.596c9.997-9.997 26.207-9.997 36.204 0l36.203 36.204c9.997 9.997 9.997 26.206 0 36.204l-294.4 294.401c-9.998 9.997-26.207 9.997-36.204-.001z"
          />
        </svg>
      </span>
    </div>
    <div
      class="c1"
      data-testid="ITR_NL"
    >
      <label
        class="c2"
      >
        Intracto NL
      </label>
      <span
        class="c3"
        data-testid="checkbox"
      >
        <svg
          color="#fff"
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          stroke-width="0"
          style="color: rgb(255, 255, 255);"
          viewBox="0 0 512 512"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M173.898 439.404l-166.4-166.4c-9.997-9.997-9.997-26.206 0-36.204l36.203-36.204c9.997-9.998 26.207-9.998 36.204 0L192 312.69 432.095 72.596c9.997-9.997 26.207-9.997 36.204 0l36.203 36.204c9.997 9.997 9.997 26.206 0 36.204l-294.4 294.401c-9.998 9.997-26.207 9.997-36.204-.001z"
          />
        </svg>
      </span>
    </div>
    <div
      class="c1"
      data-testid="FM"
    >
      <label
        class="c2"
      >
        Frontmen
      </label>
      <span
        class="c3"
        data-testid="checkbox"
      >
        <svg
          color="#fff"
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          stroke-width="0"
          style="color: rgb(255, 255, 255);"
          viewBox="0 0 512 512"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M173.898 439.404l-166.4-166.4c-9.997-9.997-9.997-26.206 0-36.204l36.203-36.204c9.997-9.998 26.207-9.998 36.204 0L192 312.69 432.095 72.596c9.997-9.997 26.207-9.997 36.204 0l36.203 36.204c9.997 9.997 9.997 26.206 0 36.204l-294.4 294.401c-9.998 9.997-26.207 9.997-36.204-.001z"
          />
        </svg>
      </span>
    </div>
  </div>
</div>
`;
